app/logic/transcript.py

```diff
@@ -10,72 +10,82 @@
+    """
+
+    otherEventsData = list(EventParticipant.select(EventParticipant, Event, Program, fn.SUM(EventParticipant.hoursEarned).alias("hoursEarned"))
+                    .join(Event)
+                    .join(ProgramEvent)
+                    .join(Program)
+                    .where(EventParticipant.user == username,
+                           Event.isTraining == False,
+                           Event.isAllVolunteerTraining == False,
+                           ((ProgramEvent.program == None) |
+                            (Program.isStudentLed == False) &
+                            (Program.isBonnerScholars == False)))
+                    .group_by(Event.term)
+                  )
+
+    otherEvents = []
+
+    for term in otherEventsData:
+        otherEvents.append([term.event.term.description, term.hoursEarned])
+
+
+    return otherEvents
+
 def getProgramTranscript(username):
     """
     Returns a Program query object containing all the programs for
     current user.
     """
-
     # Add up hours earned in a term for each program they've participated in
     programData = (ProgramEvent
         .select(Program, Event, fn.SUM(EventParticipant.hoursEarned).alias("hoursEarned"))
         .join(Program)
         .switch(ProgramEvent)
         .join(Event)
         .join(EventParticipant)
-        .where(EventParticipant.user == username, Event.isTraining == False, Program.isBonnerScholars == False))
-
-    return programData
+        .where(EventParticipant.user == username)
+        .group_by(Program, Event.term)
+        .order_by(Event.term))
 
-def getBonnerScholarEvents(username):
-    """
-    Returns a bonnerData query object containing all the Bonner events for
-    current user.
-    """
-    bonnerData = (EventParticipant
-        .select(Program, ProgramEvent, EventParticipant.event,  EventParticipant.hoursEarned)
-        .join(ProgramEvent, on=(EventParticipant.event == ProgramEvent.event))
-        .join(Program)
-        .switch(EventParticipant)
-        .join(Event)
-        .where(EventParticipant.user == username,  Event.isTraining == False, Program.isBonnerScholars == True))
+    transcriptData = {}
+    for program in programData:
+        if program.program in transcriptData:
+            transcriptData[program.program].append([program.event.term.description, program.hoursEarned])
+        else:
+            transcriptData[program.program] = [[program.event.term.description, program.hoursEarned]]
 
-    return bonnerData
+    return transcriptData
 
 def getSlCourseTranscript(username):
     """
     Returns a SLCourse query object containing all the training events for
     current user.
     """
+
     slCourses = (Course
         .select(CourseParticipant.hoursEarned, Course)
         .join(CourseParticipant, on=(Course.id == CourseParticipant.course))
-        .where(CourseParticipant.user == username).distinct())
+        .where(CourseParticipant.user == username))
```

Comment:
We shouldn't assume that the hours for each course are in one entry. We might need to sum them up into one.

Reply:
I talked with Anderson about this and since there is currently no way to add students to a course as a participant, whenever that functionality is added we should probably merge the participant instances there if duplicated exist.

Reply:
I think we want to keep them separate in the database, if they are entering them separately. It may not end up being an issue.